### DISCUSSION TITLE
[on hold] Network widget

### DISCUSSION THREAD

**Discussion by elParaguayo** [qtile/qtile#2454] [Ideas]
[Deleted original post - excessively over-engineered solution!]

**Comment by m-col** [comment]
I'm currently using the `Wlan` widget, so I'm not not interested in this. How do you foresee the roles of this widget and `Wlan` to differ? Of course the existing one is wireless only; maybe a generic network one should supercede `Wlan`?

Is it possible to have a generic way to access network information (e.g. `/sys/class/net`) that would work regardless of what network manager the system is using? Might be a bit nicer than requiring users to specify which backend they'd use and requiring the widget to interact with each manager in a different way (this is assuming the widget isn't using any manager-specific features not available in the more generic interface). The `NetGraph` widget uses `psutil`, perhaps there is a useful backend-agnostic interface there?

**Reply by elParaguayo** [reply]
To answer your first question, there were a couple of differences I wanted:
- Something a bit more comprehensive (i.e. handles both wired and wireless connections)
- Option for a graphical widget instead of the text-based `Wlan`

Thanks for the other tips. I'll take a look and see if there's anything that can be used. I would certainly love to be able to keep it really simple and backend-agnostic.

This may just be something that I use and doesn't get merged into the main repo but that's fine by me if that's the case!

**Reply by elParaguayo** [reply]
Maybe just parsing `iwconfig` is the neater answer here rather than all that mucking around with `dbus`.
